### PR TITLE
blob: use the new errors package

### PR DESCRIPTION
**Pull request by jba**
blob: use the new errors package

### CODE REVIEW

**Comment by jba** [comment]
Read only the second commit—the first is from the other PR.

**Comment by vangent** [comment]
Thanks for this, it helps put things in context. I'm leaving comments on the other PR for now (#1125).

File: blob/azureblob/azureblob.go [hunk 1]
```diff
@@ -390,32 +391,17 @@ func (b *bucket) As(i interface{}) bool {
 	return true
 }
 
-// As implements driver.ErrorAs.
-func (b *bucket) ErrorAs(err error, i interface{}) bool {
```
[the remark this answers is not in the document]
Comment (jba) [reply blob/azureblob/azureblob.go]: Done.

File: blob/blob.go [hunk 2]
```diff
@@ -785,52 +787,18 @@ func Open(ctx context.Context, urlstr string) (*Bucket, error) {
...
 // See Bucket.As for more details.
 func ErrorAs(err error, i interface{}) bool {
-	if err == nil || i == nil {
-		return false
-	}
-	if e, ok := err.(*wrappedError); ok {
-		return e.b.ErrorAs(e.err, i)
-	}
-	return false
+	return xerrors.As(err, i)
```
[the remark this answers is not in the document]
Comment (jba) [reply blob/blob.go]: Done.

File: blob/azureblob/azureblob_test.go [hunk 3]
```diff
@@ -142,14 +142,6 @@ func (verifyContentLanguage) BucketCheck(b *blob.Bucket) error {
 	return nil
 }
 
-func (verifyContentLanguage) ErrorCheck(err error) error {
```
[the remark this answers is not in the document]
Comment (jba) [reply blob/azureblob/azureblob_test.go]: Done.

**Comment by jba** [comment]
PTAL

**Comment by vangent** [comment]
Test still failing:
```
runtimevar/blobvar/blobvar.go:151: undefined: blob.IsNotExist
```

**Comment by vangent** [comment]
It looks good, I'm not approving yet because I want to see the code coverage changes after successful tests.

File: blob/blob.go [hunk 4]
```diff
@@ -41,6 +41,17 @@
...
+//
+//
+// Errors
+//
+// The errors returned from this package can be inspected in several ways:
+//
+// The Code function from gocloud.dev/gcerrors will return an error code, also
+// defined in that package, when invoked on an error.
+//
+// The As function from the package golang.org/x/xerrors can retrieve the driver
```
Comment (vangent) [inline blob/blob.go]: I would leave the `xerrors` thing off, as there's no guarantee that they will do the same thing.

Comment (jba) [reply blob/blob.go]: Done.

But this is just kicking the can down the road. When the new errors functionality lands in the standard library, we should conform to it. We can discuss another time.

**Comment by vangent** [comment]
Tests are still failing....

**Comment by jba** [comment]
PTAL

File: blob/drivertest/drivertest.go [hunk 5]
```diff
@@ -1830,9 +1831,6 @@ func testAs(t *testing.T, newHarness HarnessMaker, st AsTest) {
 	if gotErr == nil {
 		t.Fatalf("got nil error from NewReader for nonexistent key, want an error")
 	}
-	if err := st.ErrorCheck(gotErr); err != nil {
```
Comment (vangent) [inline blob/drivertest/drivertest.go]: This should not have been deleted.

Comment (jba) [reply blob/drivertest/drivertest.go]: Oops, just saw this. will fix.

Comment (jba) [reply blob/drivertest/drivertest.go]: #1167

Comment (jba) [reply blob/drivertest/drivertest.go]: Sorry, I tend to hover above the "Squash and Merge" button ready to spring into action when it goes green.

**Pull request merged**